File: practice/convert/example2_book_library/book-library/src/main/java/dev/starryeye/book_library/controller/book/request/LoanBookRequest.java
package dev.starryeye.book_library.controller.book.request;

import dev.starryeye.book_library.application.facade.input.LoanBookInput;
import jakarta.validation.constraints.NotNull;

public record LoanBookRequest(
        @NotNull(message = "bookId must not be null")
        Long bookId,
        @NotNull(message = "userId must not be null")
        Long userId
) {

    public LoanBookInput toInput() {
        return new LoanBookInput(bookId, userId);
    }
}

File: practice/convert/example2_book_library/kotlin/book-library/src/main/kotlin/dev/starryeye/book_library/controller/book/request/LoanBookRequest.kt
package dev.starryeye.book_library.controller.book.request

import dev.starryeye.book_library.application.facade.input.LoanBookInput
import jakarta.validation.constraints.NotNull

data class LoanBookRequest(
    @field:NotNull(message = "bookId must not be null")
    val bookId: Long?,
    @field:NotNull(message = "userId must not be null")
    val userId: Long?,
)

fun LoanBookRequest.toInput(): LoanBookInput =
    LoanBookInput(
        bookId = this.bookId!!,
        userId = this.userId!!,
    )